jwst/assign_wcs/util.py

```diff
@@ -88,6 +89,70 @@ def _reproject(x, y):
     return _reproject
 
 
+def compute_scale_transform(refwcs, fiducial):
+    """Compute scaling transform
+    """
+    x0, y0 = refwcs.backward_transform(*fiducial)
+
+    x1 = x0 + 1
+    y1 = y0 + 1
+    ra0, dec0 = refwcs(x0, y0)
+    ra_xdir, dec_xdir = refwcs(x1, y0)
+    ra_ydir, dec_ydir = refwcs(x0, y1)
+
+    position0 = SkyCoord(ra=ra0, dec=dec0, unit='deg')
+    position_xdir = SkyCoord(ra=ra_xdir, dec=dec_xdir, unit='deg')
+    position_ydir = SkyCoord(ra=ra_ydir, dec=dec_ydir, unit='deg')
+
+    xscale = np.abs(position0.separation(position_xdir).value)
+    yscale = np.abs(position0.separation(position_ydir).value)
+    scale = np.sqrt(xscale * yscale)
+
+    return astmodels.Scale(scale) & astmodels.Scale(scale)
```

Comment:
Also, you may just want the function to return the `scale` itself, and convert that to Astropy models in the calling function.  This gives this function and `calc_rotation_matrix` a symmetry and makes them more generally useable for both `gwcs.WCS` and `astropy.wcs.WCS` perhaps?

Reply:
I like this